gateway-ha/src/main/java/io/trino/gateway/ha/handler/ProxyUtils.java

```diff
@@ -89,31 +92,28 @@ public static String getQueryUser(String userHeader, String authorization)
         return parts.get(0);
     }
 
-    public static String extractQueryIdIfPresent(HttpServletRequest request, List<String> statementPaths)
+    public static String extractQueryIdIfPresent(
+            HttpServletRequest request,
+            List<String> statementPaths,
+            boolean requestAnalyserClientsUseV2Format,
+            int requestAnalyserMaxBodySize)
     {
         String path = request.getRequestURI();
         String queryParams = request.getQueryString();
+        String queryText = null;
         try {
-            String queryText = CharStreams.toString(new InputStreamReader(request.getInputStream()));
-            if (!isNullOrEmpty(queryText)
-                    && queryText.toLowerCase().contains("system.runtime.kill_query")) {
-                // extract and return the queryId
-                String[] parts = queryText.split(",");
-                for (String part : parts) {
-                    if (part.contains("query_id")) {
-                        Matcher matcher = EXTRACT_BETWEEN_SINGLE_QUOTES.matcher(part);
-                        if (matcher.find()) {
-                            String queryQuoted = matcher.group();
-                            if (!isNullOrEmpty(queryQuoted) && queryQuoted.length() > 0) {
-                                return queryQuoted.substring(1, queryQuoted.length() - 1);
-                            }
-                        }
-                    }
-                }
-            }
+            queryText = CharStreams.toString(new InputStreamReader(request.getInputStream()));
         }
-        catch (Exception e) {
-            log.error(e, "Error extracting query payload from request");
+        catch (IOException e) {
+            log.error(e, "Error reading request body");
```

Comment:
Why do we suppress IOException here and throw an exception for non string literal argument of `kill_query` procedure? 

The `request.getInputStream()` may contain `kill_query`, but it can go to `extractQueryIdIfPresent` in case of `IOException`, right?

Reply:
If the request body contains the `kill_query` SQL, then I already know at this point that `extractQueryIdIfPresent` will not find a query id because Trino doesn't have any endpoints with query id embedded in the `path` or `queryParams` that accept a `POST`. So falling back won't help. If I don't throw on a non-string literal argument, then this request will most likely get routed to a cluster that the target query is not running on, and the user will get a potentially confusing error message back about the query not existing. So throwing on a non-string literal allows us to fail fast with a more meaningful error.  

Do you think we should throw on `IOException` as well? I was just trying to match the existing behavior, but i'm not sure that it is correct.

Reply:
Just noting that this has been changed such that the IOException is not suppressed, and is treated as an error. If the gateway can't read the request body, Trino will not be able to either, so there is no reason to forward the request.